app/routes/user_channel.js

```diff
@@ -2,9 +2,17 @@ import BaseChannel from 'hyperchannel/routes/base_channel';
 
 export default BaseChannel.extend({
 
-  // FIXME must select acccount automatically
-  createChannelOrUserChannel: function(account, channelName){
-    return this.coms.createUserChannel(account, channelName);
+  createChannelOrUserChannel: function(account, channelId) {
+    let channel;
+    switch(account.protocol) {
```

Comment:
Is there a way this could remain unified across protocols? What's the factor that makes it impossible at the moment and is it addressable?

Reply:
That's not possible, because with IRC the direct message channel is to the username/nickname, whereas with XMPP, the channel DM is scoped to the channel (while there can also be an actual DM directly to the user).